exercises/ayomide_bamidele/ex02_oo_basics/src/Shape.h

```diff
@@ -0,0 +1,16 @@
+#ifndef SHAPE_H
+#define SHAPE_H
+#include <string>
+
+class Shape {
+private:
+  std::string type = "Shape";
+
+public:
+  virtual std::string getType() const = 0;
+  virtual int getNoOfSides() const = 0;
+  virtual double area() const = 0;
+  virtual double perimeter() const = 0;
+  ~Shape() = default;
```

Comment:
We don't need to default the constructor here as it should already be set to the default

Reply:
Okay. I didn't know that. In my defense, I was just doing what David said.

Reply:
The rules for what is declared by default are actually quite complicated depending on what you explicitly declare. The following table is a good reference for what you define vs. what the state of the special member function is:

![screen shot 2018-07-20 at 08 56 19](https://user-images.githubusercontent.com/2487781/42990593-fb8197b6-8bfa-11e8-9eed-5c66a8f2f233.png)

Reply:
And here, did you mean constructor or destructor? I thought the tilde meant that it's a destructor...

Reply:
I meant destructor. Sorry.